### PR TITLE
feat(cli): add SKILL.md auto-collection and project skills support

### PR DESCRIPTION
**Pull request by hatayama**
## Summary
- Add SKILL.md auto-collection from McpTools folders and project Editor/ folders
- Migrate CLAUDE.md documentation to `.claude/rules/` with path conditions
- Add unit tests for parseFrontmatter() and E2E tests for skills commands
- Fix get-logs E2E test stability issues related to Domain Reload
- Add link to uLoopMCP-extensions-sample project in README

## Changes
- **Skills System**: SKILL.md files are now collected from:
  - `Editor/Api/McpTools/<ToolFolder>/SKILL.md` (bundled)
  - `skill-definitions/cli-only/` (CLI-only internal skills)
  - `Assets/**/Editor/`, `Packages/**/Editor/`, `Library/PackageCache/**/Editor/` (project skills)
- **Documentation**: Migrated CLAUDE.md to `.claude/rules/cli.md` and `.claude/rules/mcp-tools.md` with paths conditions
- **Tests**: Added 11 unit tests for parseFrontmatter() and 6 E2E tests for skills commands
- **Test Stability**: Fixed get-logs test by using runCliWithRetry() and moving Domain Reload test to end

## Test plan
- [x] Unit tests pass (`npm run test:cli` in Cli~)
- [x] E2E tests pass (requires Unity running)
- [x] `uloop skills list` shows bundled skills
- [x] `uloop skills install --claude` installs skills correctly

# Skills System Enhancement and Project Skills Support

## Overview
This PR expands the skills system by adding multi-source SKILL.md collection (bundled, CLI-only, project), introduces project-local skills, migrates CLI-related CLAUDE documentation into structured rule files, and adds tests and stability fixes.

### 1. Multi-Source SKILL.md Collection
- Bundled skills: discovered from colocated SKILL.md under Packages/src/Editor/Api/McpTools/<ToolFolder>/ (generation now supports McpTools sources).
- CLI-only internal skills: placed under Packages/src/Cli~/src/skills/skill-definitions/cli-only/.
- Project skills: auto-discovered from project Editor folders:
  - Assets/**/Editor/
  - Packages/**/Editor/
  - Library/PackageCache/**/Editor/
- Deduplication prefers bundled skills over project skills; scanning respects excluded directories.

### 2. Skills Manager & Types
- New ProjectSkill type:
  - { name, dirName, content, sourcePath }
- SkillInfo gains source?: 'bundled' | 'project'.
- parseFrontmatter(content: string): Record<string, string | boolean> added and covered by unit tests.
- collectProjectSkills() added to recursively scan Editor folders and build project skills.
- getAllSkillStatuses, install/uninstall functions, and installAllSkills/uninstallAllSkills updated to accept union types (BundledSkill | ProjectSkill) and return results with bundledCount and projectCount.
- getTotalSkillCount now includes unique project skills.

### 3. Bundled-Skills Generation
- scripts/generate-bundled-skills.ts refactored to collect from MCPTOOLS_DIR (Editor/Api/McpTools/<ToolFolder>) and CLI_ONLY_DIR; computes importPath per skill and excludes frontmatter-internal skills.
- Generates Packages/src/Cli~/src/skills/bundled-skills.ts with imports referencing actual source paths and logs counts/warnings for missing sources.

### 4. CLI Behavior & Syncing
- cli.ts updated to detect unknown commands, attempt a sync of tool definitions from Unity (caches result), and dynamically register synced tools at runtime (auto-sync when unknown command encountered).
- Added commandExists helper and new async main() flow.

### 5. Documentation & Rules Migration
- CLAUDE.md content migrated into .claude/rules/cli.md and .claude/rules/mcp-tools.md with path-based conditions (e.g., Packages/src/Cli~/**).
- README and localized README_ja.md updated with a Custom Skills / uLoopMCP Extension Development section, guidance on SKILL.md placement, and a link to the uLoopMCP-extensions-sample.
- AGENTS.md simplified to emphasize the recommended namespace and condensed prior internal-skills notes.

### 6. Sample & Project SKILLs
- Added sample SKILL.md files under Assets/Editor/CustomCommandSamples/ (GetProjectInfo, GetVersion, HelloWorld).
- Added CLI-only SKILL.md for uloop-get-version in cli-only directory.
- Some prior SKILL.md files removed or relocated to match the new discovery model.

### 7. MCP Tools: UnitySearch Changes
- Removed old GetProviderDetails schema/response/tool files and replaced them with a new UnitySearchProviderDetails namespace:
  - Added UnitySearchProviderDetailsSchema, UnitySearchProviderDetailsResponse, and UnitySearchProviderDetailsTool implementing get-provider-details with similar behavior and improved model names/placement.

## Testing & Stability
- Unit tests: Added ~11 tests covering parseFrontmatter edge cases.
- E2E tests: Added 6 skills-focused tests (list, install, uninstall, counts, project inclusion) and stability improvements.
- Test stability fixes:
  - runCliWithRetry() used to handle Domain Reloads.
  - get-logs E2E setup made retry-capable and added wait constants.
  - Domain-reload-related tests reordered to run last.

## Behavior Notes & Conventions
- Skills marked internal: true in frontmatter are excluded from the bundled output.
- Install folder names use kebab-case; deduplication favors bundled skills.
- Scanning skips excluded directories (node_modules, .git, Temp, obj, Build, Builds, Logs).
- npx tsx scripts/generate-bundled-skills.ts can regenerate bundled-skills.ts from the described sources.

## Test Plan (confirmed)
- Unit tests: npm run test:cli pass.
- E2E tests require Unity; reported to pass with the updated test ordering and retry logic.
- CLI usage: uloop skills list shows bundled skills; uloop skills install --claude installs correctly.

## Impact & Public API
- Public API changes: SkillInfo gains source?, ProjectSkill and parseFrontmatter added, InstallResult now includes bundledCount and projectCount, several skills-manager function signatures widened to accept BundledSkill | ProjectSkill unions.
- generate-bundled-skills.ts added SkillMetadata.importPath field (affects generation only).
- No changes to existing public command names; new dynamic sync behavior improves discoverability.

## Notes for Reviewers
- Reviewers should check frontmatter parsing edge cases, path resolution for generated imports, deduplication priority, and the CLI unknown-command sync flow.
- Pay attention to the UnitySearch tool replacement (deletions vs. new UnitySearchProviderDetails additions) — behavior preserved but types/files relocated and renamed.

### CODE REVIEW

File: Packages/src/Editor/Api/McpTools/UnitySearchProviderDetails/UnitySearchProviderDetailsResponse.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            TotalCount = 0;
+            ActiveCount = 0;
+            InactiveCount = 0;
+            Success = false;
+            ErrorMessage = errorMessage ?? string.Empty;
+            AppliedFilter = string.Empty;
+            SortedByPriority = false;
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Editor/Api/McpTools/UnitySearchProviderDetails/UnitySearchProviderDetailsResponse.cs:56]: There are no coding conventions for null safety.
